packages/vats/src/types.js

```diff
@@ -36,18 +36,18 @@ export {};
  * @property {(key: string) => Promise<void>} reserve Mark a key as reserved; will
  * return a promise that is fulfilled when the key is updated (or rejected when
  * deleted). If the key was already set it does nothing.
- * @property {<T>( key: string, newValue: T, newAdmin?: unknown) =>
+ * @property {<T>( key: string, newValue: T, newAdmin?: NameAdmin) =>
```

Comment:
this is tricky... we can't really enforce it at runtime, but we sort of rely on it

Reply:
for all our function signatures, the param types are meant to inform the caller. the function receiving the params can't trust anything except within a guarded Exo

Reply:
right; we shouldn't rely on the caller giving the correct type at runtime; i wonder whether we do